core/domain/skill_services.py

```diff
@@ -881,8 +908,11 @@ def get_multi_skill_mastery(user_id, skill_ids):
     skill_mastery_models = user_models.UserSkillMasteryModel.get_multi(
         model_ids)
 
-    for skill_mastery_model in skill_mastery_models:
-        degrees_of_mastery.append(skill_mastery_model.degree_of_mastery)
+    for skill_id, skill_mastery_model in zip(skill_ids, skill_mastery_models):
```

Comment:
We are sure that get_multi returns None if the key doesn't exist right? Basically skill_mastery_models will have the same length as skill_ids?

Reply:
Yes, I have checked the source code for get_multi. This is also covered by test cases.